x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/deprecation/DeprecationInfoAction.java

```diff
@@ -81,24 +80,30 @@ private DeprecationInfoAction() {
         private List<DeprecationIssue> clusterSettingsIssues;
         private List<DeprecationIssue> nodeSettingsIssues;
         private Map<String, List<DeprecationIssue>> indexSettingsIssues;
-        private List<DeprecationIssue> mlSettingsIssues;
+        private Map<String, List<DeprecationIssue>> pluginSettingsIssues;
 
         public Response(StreamInput in) throws IOException {
             super(in);
             clusterSettingsIssues = in.readList(DeprecationIssue::new);
             nodeSettingsIssues = in.readList(DeprecationIssue::new);
             indexSettingsIssues = in.readMapOfLists(StreamInput::readString, DeprecationIssue::new);
-            mlSettingsIssues = in.readList(DeprecationIssue::new);
+            if (in.getVersion().before(Version.V_7_11_0)) {
```

Comment:
> Can we move the Version.V_7_11_0 to a constant? I find doing so makes the code clearer and less error-prone if we have to change the version on backport or similar.

Given that `Version.V_7_11_0` is already a constant, separating the version from the BWC serialization increases cognitive load when determining which version supports this thing. Its an extra check logical jump which would be frustrating when troubleshooting/checking what version is supported.

I would rather not replace with a constant. Especially since the almost universal pattern in the code base is to not use special constants.

Reply:
It only moves it one level (i.e. one "go to definition" click), and I find that it 1) makes changing the version during backports significantly less error prone, particularly if there are multiple version checks in the serialization, or 2) the same version is used in multiple class's serialization methods for a single logical feature.

That said, given that neither of those apply in this case, if you feel strongly about it I'm not going to pick a fight about it.